src/common/utils/helpers.ts

```diff
@@ -39,3 +40,26 @@ export const convertToListWithChildren = (_list: Array<Locator>) => {
 export const isErrorValidationType = (type: string) => VALIDATION_ERROR_TYPE.hasOwnProperty(type);
 
 export const isMacPlatform = (param: Window) => param.navigator?.userAgent.indexOf("Mac") != -1;
+
+export const generateId = (): string => {
+  return (
+    Math.random().toString().substring(2, 12) +
+    Date.now().toString().substring(5) +
+    Math.random().toString().substring(2, 12)
+  );
+};
+
+export const getElementFullXpath = (foundElement: string): string => {
+  let fullXpath = "";
+  const parser = new DOMParser();
+  const parsedElement = parser.parseFromString(foundElement, "text/html").body.firstChild;
+
+  sendMessage
+    .getElementXpath(parsedElement as Element)
+    .then((xPath: string) => {
+      if (xPath) fullXpath = xPath;
+    })
+    .catch((err: Error) => err);
+
+  return fullXpath;
```

Comment:
I guess it should be moved inside `then` or `await` should be added before `sendMessage`

Reply:
done

Reply:
fixed